src/js/angular/clustermanagement/cluster-drawing.service.js

```diff
@@ -67,3 +71,188 @@ export function moveElement(element, x, y) {
     });
 }
+
+export function createNodes(nodesDataBinding, nodeRadius) {
+    const nodeGroup = nodesDataBinding.enter()
+        .append('g')
+        .attr('id', 'node-group');
+
+    createHexagon(nodeGroup, nodeRadius);
+    nodesDataBinding.exit().remove();
+
+    nodeGroup
+        .append('text')
+        .attr('class', 'icon-any node-icon');
+
+    addHostnameToNodes(nodeGroup, nodeRadius);
+}
+
+function addHostnameToNodes(nodeElements, nodeRadius) {
+    const nodeTextHost = nodeElements.append('g');
+
+    nodeTextHost
+        .append('rect')
+        .attr('class', 'id-host-background')
+        .attr('rx', 6);
+
+    nodeTextHost
+        .append('text')
+        .attr('y', nodeRadius + 10)
+        .attr('class', 'id id-host')
+        .style('text-anchor', 'middle');
+}
+
+export function updateNodes(nodes) {
+    updateNodesIcon(nodes);
+    updateNodesClasses(nodes);
+    updateNodesHostnameText(nodes);
+}
+
+function updateNodesClasses(nodes) {
+    nodes
+        .select('.node.member')
+        .classed('leader', (node) => node.nodeState === 'LEADER')
+        .classed('follower', (node) => node.nodeState === 'FOLLOWER')
+        .classed('candidate', (node) => node.nodeState === 'CANDIDATE')
+        .classed('other', (node) => {
+            return node.nodeState !== 'LEADER' &&
+                node.nodeState !== 'FOLLOWER' &&
+                node.nodeState !== 'CANDIDATE';
+        });
+}
+
+function updateNodesIcon(nodes) {
+    nodes
+        .select('.icon-any.node-icon')
+        .text(getNodeIconType);
+}
+
+function updateNodesHostnameText(nodes) {
+    const parser = document.createElement('a');
+
+    nodes
+        .select('.id.id-host')
+        .each(function (d) {
+            d.labelNode = this;
+        })
+        .text(function (d) {
+            if (!d.endpoint) {
+                // TODO: remove this check when https://gitlab.ontotext.com/graphdb-team/graphdb/-/merge_requests/2137 is merged
+                return 'Missing endpoint';
+            } else {
+                parser.href = d.endpoint;
+                let hostname = parser.hostname;
+                if (!containsIPV4(parser.hostname)) {
+                    hostname = parser.hostname.split('.')[0];
+                }
+                return hostname + ":" + parser.port;
+            }
+        });
+
+    nodes
+        .select('.id-host-background')
+        .attr('width', function (d) {
+            return d3.select(d.labelNode).node().getBBox().width + 10;
+        })
+        .attr('height', function (d) {
+            return d3.select(d.labelNode).node().getBBox().height + 2;
+        })
+        .attr('x', function (d) {
+            return d3.select(d.labelNode).node().getBBox().x - 5;
+        })
+        .attr('y', function (d) {
+            return d3.select(d.labelNode).node().getBBox().y - 1;
+        })
+        .attr('fill', '#EEEEEE');
+}
+
+function getNodeIconType(node) {
+    if (node.nodeState === 'LEADER') {
```

Comment:
you can create enum like objects with constants for these states

Reply:
Extracted in `NodeState` and `LinkState` objects